src/pages/ListSorted.js

```diff
@@ -0,0 +1,113 @@
+import dayjs from 'dayjs';
+import React, { useContext, useState } from 'react';
+import { FirestoreCollection } from 'react-firestore';
+
+import { ListContext } from '../listContext';
+import useListToken, { getCurrentToken } from '../useListToken';
+import NavTabs from '../components/NavTabs';
+import Loading from '../components/Loading';
+import normalizeName from '../lib/normalizeName';
+
+function isLessThan24hrs(datePurchased) {
+  return dayjs().diff(dayjs(datePurchased), 'hours') <= 24;
+}
+
+//we are checking if the last date it was purchased is less than 24hrs using isLessThan24hrs function
+function isChecked(lastDatePurchased) {
+  return !!lastDatePurchased && isLessThan24hrs(lastDatePurchased);
+}
+
+const ListSort = props => {
+  const now = new Date();
+  // const today = dayjs(now);
+
+  const [filteredInput, setFilteredInput] = useState('');
+  const { shoppingList, setShoppingList, purchaseItem } = useContext(
+    ListContext,
+  );
+  const { token } = useListToken();
+
+  function handlePurchasedChange(item) {
+    // We don't want to uncheck ourselves. We should have a separate ticket for handling a mis-check
+    // What would we set datePurchased to on an uncheck? Can't be null if we've purchased or our suggestions
+    // are goofed. Maybe we live with that, or we could keep the most recent lastDatePurchased in case
+    // of a mistake. For this ticket, let's keep it simple.
+    if (!isChecked(item.lastDatePurchased)) {
+      purchaseItem(item, Date.now());
+    }
+  }
+
+  function handleFilterChange(event) {
+    setFilteredInput(event.target.value);
+  }
+
+  //5. way to clear out the filter
+
+  function filterListInput(name) {
+    return normalizeName(name).includes(normalizeName(filteredInput));
+  }
+
+  return (
+    <>
+      <FirestoreCollection
+        // Specify the path to the collection you're pulling data from
+        path="items"
+        // Sort the data
+        sort="nextExpectedPurchase:asc"
+        // Only fetch the items associated with the token saved in localStorage
+        filter={[
+          'token',
+          '==',
+          token || getCurrentToken() || 'no token set',
+          'nextExpectedPurchase',
```

Comment:
@stevelikesmusic  I think this is wrong but what I was trying to attempt is to compare `nextExpectedPurchase` with an integer and if that integer is <= 7 then I want only those items shown.